src/style/types/attribute.rs

```diff
@@ -175,7 +175,7 @@ impl Attribute {
     ///
     /// See <https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters>
     pub fn sgr(self) -> String {
-        if (self as usize) > 4 && (self as usize) < 9 {
+        if (self as usize) > 23 {
```

Comment:
can you elaborate on this check

Reply:
23 makes it difficult to work out how this code works and makes the code fragile to changes. Would this work better? It still assumes that the underline types are sequential, but makes it easier to understand what bounds are in play here.
```suggestion
        if (self as usize >= Self::DoubleUnderlined as usize) && (self as usize <= Self::Underdashed as usize) {
```

Reply:
That's safe unless you add some other variant at the end, or you move the options back somewhere within the enum like they were originally. Putting upper and lower bounds on the check makes it clear that the rationale is that these options are contiguous in the enums variants. I'd guess that having this check is unlikely to be on the hot path (and is probably even compiled out in a release build).